services/process/src/process/process.js

```
@@ -212,38 +220,32 @@ class Process {
       () => hasObjectAccess(process, requestingUserId, RESOURCE.PROCESS, PROCESS_ACTIONS.LIST_USERS)
     ])
 
-    const userIds = process.users;
-
-    const processUsers = await this.usersDB.find({ id: {$in : userIds}}).toArray();
-
-    const allUsers =  processUsers.map(entry => {
-      return {
-        username: entry.userName,
-        email: entry.email,
-        id: entry.userId
-      };
-    });
-
-    return allUsers
+    const serializedUsers = await this.serializeUsers(process)
+    return serializedUsers
   }
 
   async grantAccess({ requestingUserId, processId, data }){
     const user = await this.usersDB.findOne({email: data.email});
     const process = await this.processesDB.findOne({ id: processId });
 
+
     assertMany([
-      () => canFind(process, `Process ${processId}`), () => hasObjectAccess(process, requestingUserId, RESOURCE.PROCESS, PROCESS_ACTIONS.SHARE),
-      () => canFind(user, 'User')
+      () => canFind(process, `Process ${processId}`),
+      () => hasObjectAccess(process, requestingUserId, RESOURCE.PROCESS, PROCESS_ACTIONS.SHARE),
+      () => canFind(user, `User ${data.email}`)
     ])
 
-    const newUserId = user.id;
+    const group = process.access.find(group => group.name === data.groupName)
```

also assert now if the group actually exists instead of silent failure